build/patches/Timezone-Override.patch

```diff
@@ -0,0 +1,1379 @@
++  GetPrefProvider()->GetPrefTimezoneOverrideValue(timezone);
++}
++
++void HostContentSettingsMap::SetTimezoneOverrideValue(const std::string& timezone) {
++  GetPrefProvider()->SetPrefTimezoneOverrideValue(timezone);
++}
++
+ void HostContentSettingsMap::RecordExceptionMetrics() {
+   auto* content_setting_registry =
+       content_settings::ContentSettingsRegistry::GetInstance();
+diff --git a/components/content_settings/core/browser/host_content_settings_map.h b/components/content_settings/core/browser/host_content_settings_map.h
+--- a/components/content_settings/core/browser/host_content_settings_map.h
++++ b/components/content_settings/core/browser/host_content_settings_map.h
+@@ -336,6 +336,9 @@ class HostContentSettingsMap : public content_settings::Observer,
+     return pref_provider_;
+   }
+ 
++  void GetTimezoneOverrideValue(std::string& timezone) const;
++  void SetTimezoneOverrideValue(const std::string& timezone);
++
+  private:
+   friend class base::RefCountedThreadSafe<HostContentSettingsMap>;
+   friend class content_settings::TestUtils;
+diff --git a/components/content_settings/core/common/content_settings.cc b/components/content_settings/core/common/content_settings.cc
+--- a/components/content_settings/core/common/content_settings.cc
++++ b/components/content_settings/core/common/content_settings.cc
+@@ -90,6 +90,7 @@ constexpr HistogramValue kHistogramValue[] = {
+     {ContentSettingsType::STORAGE_ACCESS, 67},
+     {ContentSettingsType::CAMERA_PAN_TILT_ZOOM, 68},
+     {ContentSettingsType::WINDOW_PLACEMENT, 69},
++    {ContentSettingsType::TIMEZONE_OVERRIDE, 70}, // bromite #331: Customize Timezone
+ };
+ 
+ }  // namespace
+@@ -176,7 +177,8 @@ bool RendererContentSettingRules::IsRendererContentSetting(
+          content_type == ContentSettingsType::JAVASCRIPT ||
+          content_type == ContentSettingsType::CLIENT_HINTS ||
+          content_type == ContentSettingsType::POPUPS ||
+-         content_type == ContentSettingsType::MIXEDSCRIPT;
++         content_type == ContentSettingsType::MIXEDSCRIPT ||
++         content_type == ContentSettingsType::TIMEZONE_OVERRIDE; // bromite #331: Customize Timezone
+ }
+ 
+ RendererContentSettingRules::RendererContentSettingRules() {}
+diff --git a/components/content_settings/core/common/content_settings.h b/components/content_settings/core/common/content_settings.h
+--- a/components/content_settings/core/common/content_settings.h
++++ b/components/content_settings/core/common/content_settings.h
+@@ -79,6 +79,8 @@ struct RendererContentSettingRules {
+   ContentSettingsForOneType client_hints_rules;
+   ContentSettingsForOneType popup_redirect_rules;
+   ContentSettingsForOneType mixed_content_rules;
++  ContentSettingsForOneType timezone_override_rules;
++  std::string timezone_override_value;
+ };
+ 
+ namespace content_settings {
+diff --git a/components/content_settings/core/common/content_settings.mojom b/components/content_settings/core/common/content_settings.mojom
```

Comment:
Where is the protocol change? :thinking:

Reply:
sorry, here:
`array<ContentSettingPatternSource> timezone_override_rules;`
a few lines below.

Reply:
I don't see the protocol change. What is a protocol here?

Reply:
any change to the mojom files (which are the specifications) is a change to the protocol.

Reply:
@uazo what is the impact? I am not familiar. Will it break things like [remote debugging](https://developers.google.com/web/tools/chrome-devtools/remote-debugging)?

Reply:
no impact on anything.
the problem is simply that since the ipc is via pipe, they can be intercepted by applications with a proxy-like mechanism (at least in windows, in android I don't know). But my change shouldn't introduce any confidential information or permission scaling.